heartbeat/nfsserver

```diff
@@ -472,8 +472,9 @@ prepare_directory ()
 	[ -f "$fp/xtab" ] || touch "$fp/xtab"
 	[ -f "$fp/rmtab" ] || touch "$fp/rmtab"
 
-	dd if=/dev/urandom of=$fp/$STATD_DIR/state bs=1 count=4 >/dev/null 2>&1
-	[ -n "`id -u rpcuser 2>/dev/null`" -a "`id -g rpcuser 2>/dev/null`" ] && chown rpcuser.rpcuser "$fp/$STATD_DIR/state"
+	fs="$fp/$STATD_DIR/state"
+	[ -f "$fs" ] || dd if=/dev/urandom of=$fs bs=1 count=4 >/dev/null 2>&1
```

Comment:
Meanwhile, my use case is to preserve "/var/lib/nfs" in a shared storage or drbd disk for failover. Hence, intend to avoid change "/var/lib/nfs/state" whenever start nfsserver.

Not sure if this as the big concerns to other use cases. This might need further improvement. I wonder.

CC. @dpartrid

Reply:
> Meanwhile, my use case is to preserve "/var/lib/nfs" in a shared storage or drbd disk for failover. Hence, intend to avoid change "/var/lib/nfs/state" whenever start nfsserver.
> 
> Not sure if this as the big concerns to other use cases. This might need further improvement. I wonder.
> 

Go back to [the root commit 0434581](https://github.com/ClusterLabs/resource-agents/commit/0434581), I see the purpose, hence I revert my previous code change on "/var/lib/nfs/state". It is used for NFSv2 and NFSv3 actually. They are stateless protocol back then.